pkg/api/server.go

```diff
@@ -67,13 +99,51 @@ func NewServer(cfg *config.Config) (*Server, error) {
 func (srv *Server) Run(stopCh <-chan struct{}) error {
 	go func() {
 		<-stopCh
-		if err := srv.httpListener.Close(); err != nil {
-			log.Errorf("failed to close http listener: %s", err)
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		closeHttp := make(chan struct{})
+		go srv.closeHttpServer(closeHttp)
```

Comment:
The codes here are weird, we may both add a timeout for these two servers, and the closing can be parallel.

Reply:
How about this:
```go
closed := make(chan struct{}, 2)
go srv.closeHttpServer(closed)
go srv.closeAdmissionServer(closed)

ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
defer cancel()
cnt := 2
for cnt > 0 {
	select {
	case <-ctx.Done():
		log.Errorf("close servers timeout")
		return
	case <-closed:
		cnt--
		log.Debug("close a server")
	}
}
```

Reply:
should I put the timeout into the config?

Reply:
> should I put the timeout into the config?

Don't have to. And I think the code is correct.